app/models/firmware_registry.rb

```diff
@@ -31,6 +31,28 @@ def sync_fw_binaries_raw
     raise NotImplementedError, 'Must be implemented in subclass'
   end
 
+  def self.create_firmware_registry_queue(options)
+    MiqQueue.put_unless_exists(
```

Comment:
We're trying to get away from `put_unless_exists` because it is something which is not supported by any other standard queue system.